sn_node/src/node/bootstrap/join.rs

```diff
@@ -140,19 +140,20 @@ impl<'a> Joiner<'a> {
             .await?;
 
         loop {
-            let (response, sender) =
-                tokio::time::timeout(response_timeout, self.receive_join_response())
+            let (response, sender, current_sap) =
+                tokio::time::timeout(response_timeout, self.receive_join_response(target_sap))
                     .await
                     .map_err(|e| {
                         error!("Failed to receive join response: {:?}", e);
                         Error::JoinTimeout
                     })??;
+            target_sap = current_sap;
```

Comment:
If we just calculate the most up-to-date target here, we don't need to pass the sap to `receive_join_response()` or have `receive_join_response()` return the sap.
```suggestion
            target_sap = self.join_target_sap();
```

Reply:
we cannot do so, as we could waiting for a JoinResponse from sibling section

Reply:
sry, I gave it a second thought, and realize I might made a wrong statement. Will implement the change according to your comment